### ISSUE TITLE
DuckDB.close_database in julia interface doesn't release file

### ISSUE THREAD

**Issue by kobusherbst** [duckdb/duckdb#7826]
### What happens?

The file for a file-based database is not closed properly by the `close_database` function.

```julia
using DuckDB
using DBInterface

db = DuckDB.DB("C:\\Temp\\test.duckdb")
t = DBInterface.execute(db, "SELECT 1")
DuckDB.close_database(db)
```
On the second run I get the error:
```
ERROR: IO Error: Cannot open file "C:\Temp\test.duckdb": The process cannot access the file because it is being used by another process.
```

### To Reproduce

Run this code twice in the same julia session:

```julia
using DuckDB
using DBInterface

db = DuckDB.DB("C:\\Temp\\test.duckdb")
t = DBInterface.execute(db, "SELECT 1")
DuckDB.close_database(db)
```

### OS:

Windows x64

### DuckDB Version:

v0.8.0

### DuckDB Client:

julia

### Full Name:

Kobus Herbst

### Affiliation:

Africa Health Research Institute

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by bkamins** [comment]
Do you know if this has been fixed?

I have tried manually doing `DuckDB.duckdb_close` on `db.handle.handle`, but on Win11 it did not resolve the issue.

**Comment by kobusherbst** [comment]
No, it is still the case as far as I can tell. I also tried the above without success.

**Comment by kobusherbst** [comment]
This remains an issue - any plans to get this fixed - it basically makes DuckDb unusable form Julia.

**Comment by daranzolin** [comment]
I don't use Julia, but based on the code and description of the problem, I'm experiencing the same issue in R with the {duckdb} package.

**Comment by kobusherbst** [comment]
The issue has not been resolved in v1.0

**Comment by park-brian** [comment]
This looks like a Windows issue. For now, we're running everything in a Linux container (WSL2 might also work, but we haven't tried it yet). We're also using a pretty terrible workaround which looks a bit like:

```julia
using DuckDB
using DBInterface

function connect_database(filename::String)
    db = nothing
    while isnothing(db)
        try
            db = DuckDB.DB(filename)
        catch e
            println("File in use, retrying in 1 second...")
            sleep(1)
        end
    end
    return db
end

db = connect_database("C:\\Temp\\test.duckdb")
t = DBInterface.execute(db, "SELECT 1")
DuckDB.close_database(db)
```
